routes/login/login.spec.js

```diff
@@ -87,6 +89,78 @@ describe('Test /login responses', () => {
     })
   })
 
+  // Social Insurance Number Page /login/sin
+  test('it returns a 200 response for /login/sin', async () => {
+    const response = await request(app).get('/login/sin')
+    expect(response.statusCode).toBe(200)
+  })
+
+  test('it renders the h1 text for /login/sin', async () => {
+    const response = await request(app).get('/login/sin')
+
+    const $ = cheerio.load(response.text)
+    expect($('h1').text()).toEqual('Enter your Social Insurance Number (SIN)')
+  })
+
+  test('it returns a 500 response if no redirect is provided', async () => {
+    const response = await request(app).post('/login/sin')
+    expect(response.statusCode).toBe(500)
+  })
+
+  test('it reloads /login/sin with a 422 status if no sin is provided', async () => {
+    const response = await request(app)
+      .post('/login/sin')
+      .send({ redirect: '/' })
+    expect(response.statusCode).toBe(422)
+  })
+
+  describe('Error list tests', () => {
+    test('it renders the error-list for /login/sin', async () => {
+      const response = await request(app)
+        .post('/login/sin')
+        .send({ redirect: '/' })
+      const $ = cheerio.load(response.text)
+      expect($('.error-list__header').text()).toEqual('Please correct the errors on the page')
+      expect($('.error-list__list').children()).toHaveLength(1)
+      expect($('.validation-message').text()).toEqual('Your SIN should have 9 numbers')
+      expect($('#sin').attr('aria-describedby')).toEqual('sin_error')
+    })
+  })
+  /* this test seems to be doing the same thing as the one above. Do we need it?
+      test('it renders an inline error for /login/code with appropriate describedby', async () => {
+        const response = await request(app)
+          .post('/login/code')
+          .send({ redirect: '/' })
+        const $ = cheerio.load(response.text)
+        expect($('.validation-message').text()).toEqual('Access code must be 8 characters')
+        expect($('#code').attr('aria-describedby')).toEqual('code_error')
+      })
+    })
+  */
+  test('it does not allow a code more than 9 characters', async () => {
+    const response = await request(app)
+      .post('/login/sin')
+      .send({ code: '12345678910', redirect: '/' })
+    expect(response.statusCode).toBe(422)
+  })
+
+  test('it does not allow a code less than 9 characters', async () => {
+    const response = await request(app)
+      .post('/login/sin')
+      .send({ code: '12345678', redirect: '/' })
+    expect(response.statusCode).toBe(422)
+  })
+
+  test('it does not allow non-numeric characters', async () => {
+    const response = await request(app)
+      .post('/login/code')
+      .send({ code: 'A23X456@1', redirect: '/' })
+    expect(response.statusCode).toBe(422)
+  })
+
+  //todo: validate that SIN is valid and matches the user's
```

Comment:
I don't think 🤞 we need more unit tests for this. Instead, let's near the end write one/two functional tests that go through the app with some user fixture.